bots/task/test-github

```diff
@@ -32,38 +33,37 @@ import sys
 import tempfile
 import time
 import unittest
-import urlparse
+import urllib.parse
 
 BASE = os.path.dirname(__file__)
-DEVNULL = open("/dev/null", "r+")
 
-import cache
-import github
+cache = imp.load_source("cache", os.path.join(BASE, "cache.py"))
+github = imp.load_source("github", os.path.join(BASE, "github.py"))
 
 def mockServer():
     # Data used by the above handler
     data = {
         "count": 0,
     }
 
-    import BaseHTTPServer
-    class Handler(BaseHTTPServer.BaseHTTPRequestHandler):
+    import http.server
+    class Handler(http.server.BaseHTTPRequestHandler):
 
         def replyData(self, value, headers={ }, status=200):
             self.send_response(status)
-            for name, content in headers.items():
+            for name, content in list(headers.items()):
                 self.send_header(name, content)
             self.end_headers()
-            self.wfile.write(value)
-            self.wfile.close()
+            self.wfile.write(value.encode('utf-8'))
+            self.wfile.flush()
```

Comment:
If closing the file was the right thing before, why is it kept open now?

Reply:
The caller closes it.

Reply:
So before it tried to close it twice?

Reply:
Yup